internal/keystore/vault/vault.go

```diff
@@ -89,8 +91,36 @@ func Connect(ctx context.Context, c *Config) (*Store, error) {
 		}
 	}
 
-	config := vaultapi.DefaultConfig()
-	config.Address = c.Endpoint
+	// We use a custom Vault Config instead of vaultapi.DefaultConfig()
+	// since the default sets MaxIdleConnsPerHost and uses http2.ConfigureTransport.
+	// This may lead to not closing / leaking TCP connections resulting in
+	// a small but constant memory increase until running OOM.
+	dialer := &net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}
+	config := &vaultapi.Config{
+		Address: c.Endpoint,
+		HttpClient: &http.Client{
+			Timeout: 30 * time.Second,
+			Transport: &http.Transport{
+				Proxy:                 http.ProxyFromEnvironment,
+				DialContext:           dialer.DialContext,
+				ForceAttemptHTTP2:     true,
+				MaxIdleConns:          100,
```

Comment:
```suggestion
				MaxIdleConnsPerHost:          100,
```

??

Reply:
The `Transport` used in this PR does not leak TCP connections. While the `Transport` used [before](https://github.com/hashicorp/go-cleanhttp/blob/v0.5.2/cleanhttp.go#L23) does cause a resource leak. Both transports differ basically on line [36](https://github.com/hashicorp/go-cleanhttp/blob/v0.5.2/cleanhttp.go#L36).

I'm not entirely sure why the resource leak occurs. However, the using `MaxIdleConns` or `MaxIdleConnsPerHost` should make no real difference since we always talk to the same host here.